### PR TITLE
Apply defaults function

### PR DESCRIPTION
**Pull request by LucaMele**
Fixes #(issue).

# Done is when (DoD):
- [ ] I have added UI/Unit tests for my changes
- [ ] I added modified component properties to the typescript typings
- [ ] Design has been reviewed with (here name of reviewer)
- [x] My changes are accordingly to our [Best Practices](https://github.com/axa-ch/patterns-library/blob/develop/CONTRIBUTION.md#best-practices)
- [x] Old tests and eslint rule are not broken
- [x] My changes are well documented in the README and showcased in the stories 
- [x] Tested with IE

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas (focus on **WHY**)
- [ ] I have made/updated the ChangeLog Section in the README

### CODE REVIEW

File: scripts/create-a-m-o/functions.js [hunk 1]
```diff
@@ -295,11 +296,15 @@ const createFiles = (store, a, m, o, done) => () => {
         // Define properties and types
         return {
           onClick: { type: Function },
+          foo: { type: String, defualtValue: 'bar' },
```
Comment (markus-walther) [inline scripts/create-a-m-o/functions.js]: defualtValue => defaultValue

File: src/components/10-atoms/input-text/index.js [hunk 2]
```diff
@@ -161,6 +135,10 @@ class AXAInputText extends AXAPopupMixin(NoShadowDOM) {
   }
 
   get value() {
+    if (!this.nativeInput) {
```
Comment (markus-walther) [inline src/components/10-atoms/input-text/index.js:140]: small comment?

File: src/components/10-atoms/textarea/index.js [hunk 3]
```diff
@@ -93,6 +82,9 @@ class AXATextarea extends NoShadowDOM {
   }
 
   get value() {
+    if (!this.nativeInput) {
```
Comment (markus-walther) [inline src/components/10-atoms/textarea/index.js:87]: small comment?

**Review by markus-walther — DISMISSED** [review]
LGTM

File: src/utils/with-react.js [hunk 4]
```diff
@@ -1,3 +1,75 @@
...
+  return type === Array || type === Object ? JSON.parse(value) : value;
+};
+
+const applyDefaults = ceInst => {
+  const {
+    constructor: { properties },
+  } = ceInst;
+  // get all properties of the custom element and loop over each key
+  Object.keys(properties).forEach(property => {
+    // extract default value and property type found in the name of the Type
```
Comment (raphaellueckl) [inline src/utils/with-react.js]: type is uppercase, plz correct to lowercase

**Review by raphaellueckl — CHANGES_REQUESTED** [review]

File: src/utils/with-react.js (hunk 4 above)
Comment (LucaMele) [reply src/utils/with-react.js]: fixed

File: src/utils/with-react.js [hunk 5]
```diff
@@ -1,3 +1,75 @@
...
+    const propertyValue = properties[property];
+    // component author explicitly specified a default value for this property?
+    // (This allows *all* values as defaults, *including* undefined. The latter is
+    // the proper default for 'value' properties in React's controlled-component mode.)
+    const hasDefaultValue = 'defaultValue' in propertyValue;
```
Comment (01231) [inline src/utils/with-react.js]: `hasDefaultValue`is only used once. You could copy the could to row 67

**Review by 01231 — CHANGES_REQUESTED** [review]

File: src/utils/with-react.js (hunk 5 above)
Comment (LucaMele) [reply src/utils/with-react.js]: done

File: src/components/10-atoms/radio/index.js [hunk 6]
```diff
@@ -25,8 +26,8 @@ class AXARadio extends NoShadowDOM {
 
   static get properties() {
     return {
-      refId: { type: String },
-      value: { type: String },
+      refId: { type: String, defaultValue: createRefId('axa-radio-') },
```
Comment (01231) [inline src/components/10-atoms/radio/index.js:29]: would do it like in other form elements: defaultValue: \`radio-${createRefId()}\`

**Review by 01231 — CHANGES_REQUESTED** [review]

**Review by raphaellueckl — APPROVED** [review]
nice

File: src/components/10-atoms/radio/index.js (hunk 6 above)
Comment (LucaMele) [reply src/components/10-atoms/radio/index.js:29]: @NoahLiechti agree but is not part of this PR. This PR is only scoped on: Apply defaults. I created an Issue for this: #1531

**Review by 01231 — APPROVED** [review]
Nice

**Pull request merged**